scripts: support number of jobs and verbosity in west build.

tejlmand opened the pull request:
Fixes: #79099

Calling `west build --build-opt="-v" --build-opt="-j=<n>"` passes the build options to `cmake --build ... -- <build-opt>` which again is passed to the native build tool, such as ninja or make.

However, when ExternalProjects are used in CMake, such as in TF-M or sysbuild builds then those extra build options are only passed to the first image build and not those build as external projects.

CMake supports environment variables for those flags, so translate verbosity and number of jobs to those environment variables and thereby support those flags for all images when specified by the user.

tejlmand committed:
scripts: support number of jobs and verbosity in west build.

Calling `west build --build-opt="-v" --build-opt="-j=<n>"`
passes the build options to `cmake --build ... -- <build-opt>` which
again is passed to the native build tool, such as ninja or make.

However, when ExternalProjects are used in CMake, such as in TF-M or
sysbuild builds then those extra build options are only passed to the
first image build and not those build as external projects.

CMake supports environment variables for those flags, so translate
verbosity and number of jobs to those environment variables and thereby
support those flags for all images when specified by the user.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

carlescufi reviewed (commented):
This is great @tejlmand, thank you.

I wonder, for those using CMake and ninja directly, would it make sense to document in the sysbuild doc that they need to use the `CMAKE_BUILD_PARALLEL_LEVEL` and `VERBOSE` env vars?

tejlmand commented:
> I wonder, for those using CMake and ninja directly, would it make sense to document in the sysbuild doc that they need to use the `CMAKE_BUILD_PARALLEL_LEVEL` and `VERBOSE` env vars?

that's a good idea.

tejlmand committed:
doc: document CMAKE_BUILD_PARALLEL_LEVEL and VERBOSE env variables

Sysbuild may include several images and when using ninja as build tools
then build options like `ninja -v` or `ninja -j <n>` is only consumed
by the top-level build folder, which is sysbuild itself.

CMake supports setting number of jobs or verbosity using environment
variables.

Document this feature in the Building with sysbuild section.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

tejlmand commented:
> that's a good idea.

@carlescufi done
